tests/Providers/Mistral/MistralOCRTest.php

```diff
@@ -0,0 +1,277 @@
+
+TIP
+It is recommended to set either temperature or topP, but not both.
+usingTopP
+Nucleus sampling.
+The value is passed through to the provider. The range depends on the provider and model. For most providers, nucleus sampling is a number between 0 and 1. E.g. 0.1 would mean that only tokens with the top $10 \%$ probability mass are considered.
+
+TIP
+It is recommended to set either temperature or topP, but not both.
+
+```
+withClientOptions
+```
+
+Under the hood we use L aravel\'s HTTP client. You can use this method to pass any of Guzzles request options e.g. ->withClientOptions([\'timeout\' => 30]) .
+withClientRetry
+
+Under the hood we use L aravel\'s HTTP client. You can use this method to set retries e.g. ->withClientRetry(3, 100) .
+usingProviderConfig
+
+This allows for complete or partial override of the providers configuration. This is great for multi-tenant applications where users supply their own API keys. These values are merged with the original configuration allowing for partial or complete config override.
+
+# Response Handling 
+
+The response object provides rich access to the generation results:
+
+```
+use Prism\Prism\Prism;
+use Prism\Prism\Enums\Provider;
+$response = Prism::text()
+    ->using(Provider::Anthropic, \'claude-3-5-sonnet-20241022\')
+    ->withPrompt(\'Explain quantum computing.\')
+    ->asText();
+    // Access the generated text
+    echo $response->text;
+    // Check why the generation stopped
+    echo $response->finishReason->name;
+    // Get token usage statistics
+    echo "Prompt tokens: {$response->usage->promptTokens}";
+    echo "Completion tokens: {$response->usage->completionTokens}";
+    // For multi-step generations, examine each step
+    foreach ($response->steps as $step) {
+        echo "Step text: {$step->text}";
+
+```
+        echo "Step tokens: {$step->usage->completionTokens}";
+    }
+    // Access message history
+    foreach ($response->responseMessages as $message) {
+        if ($message instanceof AssistantMessage) {
+            echo $message->content;
+        }
+    }
+```
+
+
+# Finish Reasons 
+
+FinishReason: :Stop;
+FinishReason: : Length;
+FinishReason: : ContentFilter;
+FinishReason: : ToolCalls;
+FinishReason: :Error;
+FinishReason: :Other;
+FinishReason: :Unknown;
+
+## Error Handling
+
+Remember to handle potential errors in your generations:
+
+```
+php
+use Prism\Prism\Prism;
+use Prism\Prism\Enums\Provider;
+use Prism\Prism\Exceptions\PrismException;
+use Throwable;
+try {
+    $response = Prism::text()
+        ->using(Provider::Anthropic, \'claude-3-5-sonnet-20241022\')
+        ->withPrompt(\'Generate text...\')
+        ->asText();
+    } catch (PrismException $e) {
+    Log::error(\'Text generation failed:\', [\'error\' => $e->getMessage()]);
+    } catch (Throwable $e) {
+    Log::error(\'Generic error:\', [\'error\' => $e->getMessage()]);
+```
+
+Previous page
+Configuration
+Next page
+Streaming Ouput');
```

Comment:
How do we feel about making just some more generic assertions to help keep things tidy?

```suggestion
  expect($object->toText())->toContain('# Text Generation');
    expect($object->toText())->toContain('## Basic Text Generation');
    expect($object->toText())->toContain('## System Prompts and Context');
    expect($object->toText())->toContain('# Message Chains and Conversations');
    expect($object->toText())->toContain('## Message Types');
    expect($object->toText())->toContain('# Generation Parameters');
    expect($object->toText())->toContain('# Response Handling');
    expect($object->toText())->toContain('# Finish Reasons');
    expect($object->toText())->toContain('## Error Handling');
```

Reply:
Totally agree, way cleaner this way.